api/src/Service/JWT/JWTService.php

```diff
@@ -0,0 +1,81 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Service\JWT;
+
+use App\Entity\User;
+use App\Service\SecretManagerService;
+use Firebase\JWT\JWT;
+use Firebase\JWT\Key;
+use Psr\Log\LoggerInterface;
+
+class JWTService
+{
+    public const JKU_URL_TEMPLATE = '%s/v2/.well-known/jwks.json';
+    public const JWT_ISS = 'digideps';
+
+    public function __construct(
+        private SecretManagerService $secretManager,
+        private LoggerInterface $logger,
+        private string $frontendHost
+    ) {
+    }
+
+    public function createNewJWT(User $user)
+    {
+        $privateKey = base64_decode($this->secretManager->getSecret(SecretManagerService::PRIVATE_JWT_KEY_BASE64_SECRET_NAME));
+        $publicKey = base64_decode($this->secretManager->getSecret(SecretManagerService::PUBLIC_JWT_KEY_BASE64_SECRET_NAME));
+
+        $kid = openssl_digest($publicKey, 'sha256');
+        $payload = [
+            'aud' => 'registration_service',
```

Comment:
I've started a very early draft of an OPG JWT standard over in this [Google doc](https://docs.google.com/document/d/1AccZBj-YZdhRk3AG_TlV9Bdyf17HaGW7qmH01_igfr8/edit?usp=sharing). In particular I think the bits about using urns for subject and audiences might be useful to shape this code.

Reply:
Thanks Matt - that's useful to see how it would work in practice and we can implement that here. I guess we'd need to think through how we're going to standardise the urns and make available some way to access a list of existing ones to use in services rather than hard coding.

One thing I've been thinking through is the use case I'm implementing this change for; services without a concept of a user that need to interact with digideps. My initial thought was to have some central token store where you register the service that is going to have an integration and then receive some type of API key (or the JWT itself) that you can pull into or set in the service. A quick iteration of this would be to manually or automatically generate a JWT and store in secret manager to be pulled into a service.

Can you see any issues with this approach?

Reply:
I seem to remember this is very similar to a pattern that @andrewpearce-digital had started on in the metrics service. i.e. a job that periodically regenerates a service JWT and pushes to secrets manager.

Although to be honest if we can get all services to generate a JWT the need for service level accounts is reduced as every end user could be issued a claim appropriate for  interacting with multiple services.